src/TreeNode.java

```diff
@@ -0,0 +1,18 @@
+public class TreeNode {
+    int val;
+    TreeNode left;
+    TreeNode right;
+
+    TreeNode() {
```

Comment:
do we really need empty constructor?

Reply:
this part of the code is presented by an example on leetcode